File: baekjoon/shim/Python/단계별로_풀어보기/브루트포스/n_1018.py
n, m = map(int, input().split())
board = []
cnt = []
for i in range(n):
    board.append(input())
for a in range(n-7):
    for b in range(m-7):
        n1,n2 = 0,0
        for i in range(a, a+8):
            for j in range(b, b+8):
                if (i + j) % 2 == 0:
                    if board[i][j] != 'W':
                        n1 += 1
                    if board[i][j] != 'B':
                        n2 += 1
                else:
                    if board[i][j] != 'B':
                        n1 += 1
                    if board[i][j] != 'W':
                        n2 += 1                 
        cnt.append(min(n1, n2))
print(min(cnt))

File: baekjoon/shim/Java/단계별로_풀어보기/브루트포스/n_1018.java
package Study_2022.baekjoon.shim.Java.단계별로_풀어보기.브루트포스;

import java.io.*;

public class n_1018 {
	public static int func(int row, int col, String[] board) {
		String orgBoard[] = {"WBWBWBWB", "BWBWBWBW"};
		int whiteSol = 0;
		for(int i=0; i<8; i++) {
			int r = row + i;
			for(int j=0; j<8; j++) {
				int c = col + j;
				if(board[r].charAt(c) != orgBoard[r % 2].charAt(j))
					whiteSol++;
			}
		}
		return Math.min(whiteSol, 64 - whiteSol);
	}
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		String[] st = br.readLine().split(" ");
		int r = Integer.parseInt(st[0]), c = Integer.parseInt(st[1]),min = Integer.MAX_VALUE;
		String board[] = new String[r];
		for(int i=0; i<r; i++)
			board[i] = br.readLine();
		for(int i=0; i<r - 7; i++) {
			for(int j=0; j<c - 7; j++) {
				int num = func(i, j, board);
				if(min > num)
					min = num;
			}
		}
		System.out.println(min);
	}
}
